src/Database/DBStructure.php

```diff
@@ -996,7 +996,8 @@ public static function definition() {
 				"indexes" => array(
 						"PRIMARY" => array("id"),
 						"contactid" => array("contact-id"),
-						"gid_contactid" => array("gid", "contact-id"),
+						"gid" => array("gid"),
```

Comment:
Sorry to be nitpicking, but you don't need this line, the next line is sufficient - and you have to increase the database version number constants in boot.php (DB_UPDATE_VERSION) and update.php (UPDATE_VERSION) so that the system will perform an update.

Reply:
Can I remove the `contact-id` index as well then?

Reply:
Indexes are read from left to right. Means that a "gid, contact-id" always contains the "gid" as well. But a query to the "contact-id" alone would be "cost intense". Means: When there queries to that table that have a condition containing the "contact-id" but not the "gid" then we still need that index.

Best is always to activate the index usage statistics on the SQL server. There we can see which indexes are used how often.

Reply:
I didn't know the left to right rule. I usually am creating more indices than I probably need, because the penalty for having too many indices is always smaller than having too few. As a rule I'm creating an index on any column I'm having a condition or a join on.

Reply:
MySQL is really bad in combining different indexes, so a single combined one is better. Index optimization is a really interesting topic.

Reply:
Gotcha. I also use combined indices for n:m tables like `group_member` and recurring combined conditions but I didn't know the left to right rule.